Library/Homebrew/rubocops/urls_cop.rb

```diff
@@ -0,0 +1,102 @@
+                                            %r{^http://download\.savannah\.gnu\.org/},
+                                            %r{^http://download-mirror\.savannah\.gnu\.org/},
+                                            %r{^http://[^/]*\.apache\.org/},
+                                            %r{^http://code\.google\.com/},
+                                            %r{^http://fossies\.org/},
+                                            %r{^http://mirrors\.kernel\.org/},
+                                            %r{^http://(?:[^/]*\.)?bintray\.com/},
+                                            %r{^http://tools\.ietf\.org/},
+                                            %r{^http://launchpad\.net/},
+                                            %r{^http://github\.com/},
+                                            %r{^http://bitbucket\.org/},
+                                            %r{^http://anonscm\.debian\.org/},
+                                            %r{^http://cpan\.metacpan\.org/},
+                                            %r{^http://hackage\.haskell\.org/},
+                                            %r{^http://(?:[^/]*\.)?archive\.org},
+                                            %r{^http://(?:[^/]*\.)?freedesktop\.org},
+                                            %r{^http://(?:[^/]*\.)?mirrorservice\.org/}])
+
+          # CPAN pattern
+          cpan_pattern = %r{^http://search\.mcpan\.org/CPAN/(.*)}i
+
+          # GNOME pattern
+          gnome_pattern = %r{^(http|ftp)://ftp\.gnome\.org/pub/gnome/(.*)}i
+
+          # DEBIAN pattern
+          debian_pattern = %r{^git://anonscm\.debian\.org/users/(.*)}i
+
+          urls = find_every_method_call_by_name(body_node, :url)
+          mirrors = find_every_method_call_by_name(body_node, :mirror)
+
+          audit_urls(urls, gnu_pattern) do |match_obj, url|
+            problem "Please use \"https://ftp.gnu.org/gnu/#{match_obj[1]}\" instead of #{url}."
+          end
+
+          audit_urls(urls, fossies_pattern) do |_, _|
+            problem "Please don't use fossies.org in the url (using as a mirror is fine)"
+          end
+
+          audit_urls(mirrors, /.*/) do |_, mirror|
+            urls.each do |url|
+              url_string = string_content(parameters(url).first)
+              if url_string.eql?(mirror)
+                problem "URL should not be duplicated as a mirror: #{url_string}"
+              end
+            end
+          end
+
+          urls += mirrors
+
+          audit_urls(urls, http_urls_pattern) do |_, url|
+            problem "Please use https:// for #{url}"
+          end
+
+          audit_urls(urls, cpan_pattern) do |match_obj, url|
+            problem "#{url} should be `https://cpan.metacpan.org/#{match_obj[1]}`"
+          end
+
+          audit_urls(urls, gnome_pattern) do |match_obj, url|
+            problem "#{url} should be `https://download.gnome.org/#{match_obj[2]}`"
```

Comment:
Let's define e.g. `gnome_pattern` immediately above where it is called so it's obvious what regex group `match[2]` refers to.